internal/validation/validate.go

```diff
@@ -128,7 +127,8 @@ func (v *APIRule) validateService(attributePath string, api *gatewayv1beta1.APIR
 
 	for namespace, services := range v.ServiceBlockList {
 		for _, svc := range services {
-			if svc == *api.Spec.Service.Name && namespace == api.ObjectMeta.Namespace {
+			serviceNamespace := helpers.FindServiceNamespace(api, nil)
```

Comment:
`nilpointer dereference` is most likely not possible to occur, but I would handle it just to be  100% sure